execute/table/copy.go

```diff
@@ -89,13 +102,20 @@ func (tb *tableBuffer) BufferN() int {
 }
 
 func (tb *tableBuffer) Copy() flux.BufferedTable {
-	for i := tb.i; i < len(tb.buffers); i++ {
+
+	// TODO: make this panic instead of just logging when running tests.
+	//  Arrow has a `debug.Assert` thing which will noop unless the assert build
+	//  tag is set. We could do something similar.
+	if atomic.LoadInt32(&tb.used) == 1 {
+		log.Println("tried to copy an already used tableBuffer")
```

Comment:
I saw a random usage of `log.Println()` over in `compiler/runtime.go` :shrug:

I just didn't want to pile on adding something like that with this PR. Would rather bikeshed it separately if needed.

Reply:
I was hesitant to add this immediately since verification seemed tricky. Added with b8c5a74786e86e93a54da5e2703323023db94eb1

I weakly tested by flipping the assertion condition and adding/removing the build tag in the Makefile to make sure panics happened or didn't happen when we wanted.